commonv5/ts/sonarqube/__tests__/Analysis-test.ts

```diff
@@ -1,101 +1,90 @@
+const MOCKED_PROJECT_STATUS_ERROR = {
+  status: "ERROR",
+  conditions: [
+    {
+      status: "ERROR",
+      metricKey: "bugs",
+      comparator: "GT",
+      errorThreshold: "0",
+      actualValue: "1",
+    },
+  ],
+};
+const MOCKED_PROJECT_STATUS_SUCCESS = {
+  status: "SUCCESS",
+  conditions: [],
+};
+const MOCKED_ANALYSIS_RESULT: AnalysisResult = {
+  dashboardUrl: "https://dashboard.url",
+  projectName: null,
+  metrics: MOCKED_METRICS,
+  warnings: [],
+};
 
 jest.mock("azure-pipelines-task-lib/task", () => ({
   debug: jest.fn(),
   error: jest.fn(),
   getHttpProxyConfiguration: jest.fn().mockImplementation(() => null),
 }));
 
-const METRICS = new Metrics([{ key: "bugs", name: "Bugs", type: "INT" }]);
-const ENDPOINT = new Endpoint(EndpointType.SonarQube, { url: "https://endpoint.url" });
-const GET_ANALYSIS_DATA = {
-  analysisId: "analysisId",
-  dashboardUrl: "https://dashboard.url",
-  endpoint: ENDPOINT,
-  metrics: METRICS,
-  warnings: [],
-};
-
-beforeEach(() => {
-  (get as jest.Mock<any>).mockClear();
-});
+it("should generate an analysis status with error", () => {
+  const analysis = new Analysis(
+    MOCKED_ENDPOINT.type,
+    MOCKED_PROJECT_STATUS_ERROR,
+    MOCKED_ANALYSIS_RESULT,
+  );
 
-it("should generate an analysis status with error", async () => {
-  const analysis = await Analysis.getAnalysis(GET_ANALYSIS_DATA);
-  expect(get).toHaveBeenCalledWith(ENDPOINT, "/api/qualitygates/project_status", true, {
-    analysisId: "analysisId",
-  });
-  expect(analysis.status).toBe("ERROR");
   expect(analysis.getFailedConditions()).toHaveLength(1);
   expect(analysis.getHtmlAnalysisReport()).toMatchSnapshot();
 });
 
-it("should generate a green analysis status", async () => {
-  (get as jest.Mock<any>).mockImplementationOnce(() =>
-    Promise.resolve({ projectStatus: { status: "SUCCESS", conditions: [] } }),
+it("should generate a green analysis status", () => {
+  const analysis = new Analysis(
+    MOCKED_ENDPOINT.type,
+    MOCKED_PROJECT_STATUS_SUCCESS,
+    MOCKED_ANALYSIS_RESULT,
   );
 
-  const analysis = await Analysis.getAnalysis(GET_ANALYSIS_DATA);
-  expect(get).toHaveBeenCalledWith(ENDPOINT, "/api/qualitygates/project_status", true, {
-    analysisId: "analysisId",
-  });
-  expect(analysis.status).toBe("SUCCESS");
   expect(analysis.getFailedConditions()).toHaveLength(0);
   expect(analysis.getHtmlAnalysisReport()).toMatchSnapshot();
 });
 
-it("should not fail when metrics are missing", async () => {
-  const analysis = await Analysis.getAnalysis({
-    ...GET_ANALYSIS_DATA,
+it("should not fail when metrics are missing", () => {
+  const analysis = new Analysis(MOCKED_ENDPOINT.type, MOCKED_PROJECT_STATUS_ERROR, {
+    ...MOCKED_ANALYSIS_RESULT,
     dashboardUrl: undefined,
     metrics: undefined,
   });
-  expect(get).toHaveBeenCalledWith(ENDPOINT, "/api/qualitygates/project_status", true, {
-    analysisId: "analysisId",
-  });
-  expect(analysis.status).toBe("ERROR");
```

Comment:
These assertions were duplicated in each test. Moved it to `utils-test,tsx` as they are testing `fetchProjectStatus`